tests/testsuite/freshness.rs

```diff
@@ -2381,8 +2385,21 @@ fn linking_interrupted() {
 
     // Interrupt the child.
     child.kill().unwrap();
-    // Note: rustc and the linker are still running, let them exit here.
-    conn.write(b"X").unwrap();
+    child.wait().unwrap();
+    // Note: rustc and the linker may still be running because we didn't kill
+    // the entire process group. Normally, when a user hits Ctrl-C, everything
+    // is killed. However, setting up process groups in a cross-platform test
+    // is a pain, and there's no easy way to know when everything has been
```

Comment:
FWIW we've [already got test code that does this](https://github.com/rust-lang/cargo/blob/6a2d62fb402d1cd47d8eadf890522fdba3044b48/tests/testsuite/death.rs#L51-L132), perhaps it wouldn't be too bad to refactor that so it could be used here too?